rat/ops.py

```diff
@@ -21,6 +24,8 @@ def __str__(self):
 @dataclass(frozen=True)
 class RealConstant(Expr):
     value: float
+    line_index: int = -1  # line index of the original model code
+    column_index: int = -1  # column index of the original model code
```

Comment:
I think we can put the defaults in the base class (and make Expr a dataclass). There's a trick to make this work, but I think it should be possible: https://stackoverflow.com/a/58525728.

Reply:
Sounds good. I was quite frustrated having to copy-paste those two lines repeatedly.

Reply:
Cool beans. It's some Python 3.10 stuff that makes this easy: https://stackoverflow.com/a/69822584